libraries/program-error/tests/mod.rs

```diff
@@ -0,0 +1,135 @@
+            <bench::ExampleError as DecodeError<decode::ExampleError>>::type_of(),
+        );
+    }
+    // `#[derive(PrintProgramError)]`
+    #[test]
+    fn test_derive_print_program_error() {
+        use std::sync::Once;
+        static ONCE: Once = Once::new();
+
+        ONCE.call_once(|| {
+            solana_sdk::program_stubs::set_syscall_stubs(Box::new(SyscallStubs {}));
+        });
+        // `Into<ProgramError>`
+        assert_eq!(
+            Into::<ProgramError>::into(bench::ExampleError::MintHasNoMintAuthority),
+            Into::<ProgramError>::into(print::ExampleError::MintHasNoMintAuthority),
+        );
+        assert_eq!(
+            Into::<ProgramError>::into(bench::ExampleError::IncorrectMintAuthority),
+            Into::<ProgramError>::into(print::ExampleError::IncorrectMintAuthority),
+        );
+        // `DecodeError<T>`
+        assert_eq!(
+            <bench::ExampleError as DecodeError<bench::ExampleError>>::type_of(),
+            <bench::ExampleError as DecodeError<print::ExampleError>>::type_of(),
+        );
+        // `PrintProgramError`
+        set_expected_data("Mint has no mint authority".as_bytes().to_vec());
+        PrintProgramError::print::<print::ExampleError>(
+            &print::ExampleError::MintHasNoMintAuthority,
+        );
+        set_expected_data(
+            "Incorrect mint authority has signed the instruction"
+                .as_bytes()
+                .to_vec(),
+        );
+        PrintProgramError::print::<print::ExampleError>(
+            &print::ExampleError::IncorrectMintAuthority,
+        );
+    }
+
+    // `#[spl_program_error]`
+    #[test]
+    fn test_spl_program_error() {
+        use std::sync::Once;
+        static ONCE: Once = Once::new();
+
+        ONCE.call_once(|| {
+            solana_sdk::program_stubs::set_syscall_stubs(Box::new(SyscallStubs {}));
+        });
+        // `Into<ProgramError>`
+        assert_eq!(
+            Into::<ProgramError>::into(bench::ExampleError::MintHasNoMintAuthority),
+            Into::<ProgramError>::into(spl::ExampleError::MintHasNoMintAuthority),
+        );
+        assert_eq!(
+            Into::<ProgramError>::into(bench::ExampleError::IncorrectMintAuthority),
+            Into::<ProgramError>::into(spl::ExampleError::IncorrectMintAuthority),
+        );
+        // `DecodeError<T>`
+        assert_eq!(
+            <bench::ExampleError as DecodeError<bench::ExampleError>>::type_of(),
+            <bench::ExampleError as DecodeError<spl::ExampleError>>::type_of(),
+        );
+        // `PrintProgramError`
+        set_expected_data("Mint has no mint authority".as_bytes().to_vec());
+        PrintProgramError::print::<spl::ExampleError>(&spl::ExampleError::MintHasNoMintAuthority);
+        set_expected_data(
+            "Incorrect mint authority has signed the instruction"
+                .as_bytes()
+                .to_vec(),
+        );
```

Comment:
Looks like this was missing the check
```suggestion
        set_expected_data(
            "Incorrect mint authority has signed the instruction"
                .as_bytes()
                .to_vec(),
        );
        PrintProgramError::print::<print::ExampleError>(
            &print::ExampleError::IncorrectMintAuthority,
        );
```